src/Perl6/Metamodel/RolePunning.nqp

```diff
@@ -66,11 +66,13 @@ role Perl6::Metamodel::RolePunning {
             $!pun := self.make_pun($obj);
             $!made_pun := 1;
         }
-        unless nqp::can($!pun, $name) {
+
+        my $meth := $!pun.HOW.find_method($!pun, $name);
```

Comment:
This should probably forward any named parameters given.

Reply:
I suppose that will make it more consistent. Thanks for the suggestion!